django-backend/fecfiler/reports/views.py

```diff
@@ -0,0 +1,187 @@
+    When(report_code="M11", then=Value("NOVEMBER 20 (M11)")),
+    When(report_code="M12", then=Value("DECEMBER 20 (M12)")),
+)
+
+
+def get_status_mapping():
+    """returns Django Case that determines report status based on upload submission"""
+    in_progress = Q(upload_submission__fec_status=None) | Q(upload_submission=None)
+    submitted = Q(
+        upload_submission__fecfile_task_state__in=[
+            FECSubmissionState.INITIALIZING,
+            FECSubmissionState.CREATING_FILE,
+            FECSubmissionState.SUBMITTING,
+        ]
+    ) | Q(upload_submission__fec_status=FECStatus.PROCESSING)
+    success = Q(upload_submission__fec_status=FECStatus.ACCEPTED)
+    failed = Q(upload_submission__fecfile_task_state=FECSubmissionState.FAILED) | Q(
+        upload_submission__fec_status=FECStatus.REJECTED
+    )
+
+    return Case(
+        When(in_progress, then=Value("In progress")),
+        When(submitted, then=Value("Submission pending")),
+        When(success, then=Value("Submission success")),
+        When(failed, then=Value("Submission failure")),
+    )
+
+
+class ReportViewSet(CommitteeOwnedViewSet):
+    """
+    This viewset automatically provides `list`, `create`, `retrieve`,
+    `update` and `destroy` actions.
+
+    Note that this ViewSet inherits from CommitteeOwnedViewSet
+    The queryset will be further limited by the user's committee
+    in CommitteeOwnedViewSet's implementation of get_queryset()
+    """
+
+    queryset = (
+        Report.objects.annotate(report_code_label=report_code_label_mapping)
+        .annotate(report_status=get_status_mapping())
+        .all()
+    )
+
+    serializer_class = ReportSerializer
+    filter_backends = [filters.OrderingFilter]
+    ordering_fields = [
+        "form_type",
+        "upload_submission__fec_status",
+        "submission_status",
+    ]
+    ordering = ["form_type"]
+
+    # Allow requests to filter reports output based on report type by
+    # passing a query parameter
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        report_type_filters = self.request.query_params.get("report_type")
+        if report_type_filters is not None:
+            report_type_list = report_type_filters.split(",")
+            # All transactions are included by default, here we remove those
+            # that are not identified in the schedules query param
+            if "f3x" not in report_type_list:
+                queryset = queryset.filter(form_3x__isnull=True)
+            if "f24" not in report_type_list:
+                queryset = queryset.filter(form_24__isnull=True)
+            if "f99" not in report_type_list:
+                queryset = queryset.filter(form_99__isnull=True)
+        return queryset
+
+    @action(
+        detail=True,
+        methods=["post"],
+        url_name="amend",
+    )
+    def amend(self, request, pk):
+        report = self.get_object()
+        report.form_type = "F3XA"
```

Comment:
not a right now thing, but we'll have to update it for the next type that we can amend